Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/util/ConnectionHelper.java

```diff
@@ -143,4 +144,33 @@ protected PropertyAccessor getPropertyAccessor() {
         return PropertyAccessor.getInstance();
     }
 
+    public Map<String, String> getOrgNameRemoteHcidExternalEntities() {
+        HashMap<String, String> organizationMap = new HashMap<>();
+        List<BusinessEntity> externalEntityList = new ArrayList<>(getExternalEntitiesMap().values());
+        if (NullChecker.isNotNullish(externalEntityList)) {
+            for (BusinessEntity businessEntity : externalEntityList) {
+                List<KeyedReference> keyedReference = getKeyedReference(businessEntity);
+                if (NullChecker.isNotNullish(keyedReference)) {
+                    organizationMap.put(getEntityName(businessEntity), keyedReference.get(0).getKeyValue());
+                }
+            }
+        }
+        return organizationMap;
+    }
+
+    public Map<String, String> getRemoteHcidOrgNameMap() {
+        HashMap<String, String> organizationMap = new HashMap<>();
+        List<BusinessEntity> entities = getAllBusinessEntities();
+        if (NullChecker.isNotNullish(entities)) {
+            for (BusinessEntity businessEntity : entities) {
+                List<KeyedReference> keyedReference = getKeyedReference(businessEntity);
+                if (NullChecker.isNotNullish(keyedReference)
+                    && NullChecker.isNotNullishIgnoreSpace(keyedReference.get(0).getKeyValue())) {
+                    organizationMap.put(HomeCommunityMap.getHomeCommunityWithoutPrefix(
+                        keyedReference.get(0).getKeyValue()), getEntityName(businessEntity));
+                }
+            }
+        }
+        return organizationMap;
```

Comment:
I think there is a need for 2 maps here. One map has Key= OrgName, Value= remoteHcid without the external entities. Other map has Key=remoteHcid, Value = Orgname with the external entity as well. 
However if we need to fix it we can do it as part of FHAC-747